pallets/claim/src/lib.rs

```diff
@@ -164,89 +180,172 @@ decl_error! {
-	pub struct Module<T: Config> for enum Call where origin: T::Origin {
-		// Errors must be initialized if they are used by the pallet.
-		type Error = Error<T>;
-		fn deposit_event() = default;
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
 
-		#[weight = 0]
-		pub fn change_relayer(origin, new_relayer: T::AccountId) -> dispatch::DispatchResult {
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight(0 + T::DbWeight::get().reads_writes(1,1))]
+		pub fn change_relayer(
+			origin: OriginFor<T>,
+			new_relayer: T::AccountId,
+		) -> DispatchResultWithPostInfo {
 			ensure_root(origin)?;
 			let old_relayer = Relayer::<T>::get();
-			ensure!(Some(&new_relayer) != old_relayer.as_ref(), Error::<T>::RelayerNotChanged);
-			Relayer::<T>::put(&new_relayer);
-			Self::deposit_event(RawEvent::RelayerChanged(new_relayer));
-			Ok(())
+			if old_relayer.is_some() {
+				ensure!(
+					Some(&new_relayer) != old_relayer.unwrap().as_ref(),
+					Error::<T>::RelayerNotChanged
+				);
+			}
+
+			Relayer::<T>::put(Some(&new_relayer));
+			Self::deposit_event(Event::RelayerChanged(new_relayer));
+			Ok(().into())
 		}
 
-		#[weight = 0]
-		pub fn store_erc20_burned_transactions(origin, height: u64, claims:Vec<(EthereumTxHash, EthereumAddress, BalanceOf<T>)>) -> dispatch::DispatchResult {
+		#[pallet::weight(0 + T::DbWeight::get().reads_writes(1,1))]
+		pub fn store_erc20_burned_transactions(
+			origin: OriginFor<T>,
+			height: u64,
+			claims: Vec<(EthereumTxHash, EthereumAddress, BalanceOf<T>)>,
+		) -> DispatchResultWithPostInfo {
 			let who = ensure_signed(origin)?;
 			let relayer = Relayer::<T>::get();
 			ensure!(relayer.is_some(), Error::<T>::NoRelayer);
-			ensure!(Some(&who) == relayer.as_ref(), Error::<T>::CallerNotRelayer);
+			ensure!(
+				Some(&who) == relayer.unwrap().as_ref(),
+				Error::<T>::CallerNotRelayer
+			);
 			// check first
 			for (eth_tx_hash, _, _) in claims.iter() {
-				ensure!(!BurnedTransactions::<T>::contains_key(&eth_tx_hash), Error::<T>::TxHashAlreadyExist);
+				ensure!(
+					!BurnedTransactions::<T>::contains_key(&eth_tx_hash),
+					Error::<T>::TxHashAlreadyExist
+				);
 			}
 			for (eth_tx_hash, eth_address, erc20_amount) in claims.iter() {
-				BurnedTransactions::<T>::insert(&eth_tx_hash, (eth_address.clone(), erc20_amount.clone()));
-				ClaimState::insert(&eth_tx_hash, false);
-				Self::deposit_event(RawEvent::ERC20TransactionStored(who.clone(), *eth_tx_hash, *eth_address, *erc20_amount));
+				BurnedTransactions::<T>::insert(
+					&eth_tx_hash,
+					(eth_address.clone(), erc20_amount.clone()),
+				);
+				ClaimState::<T>::insert(&eth_tx_hash, false);
+				Self::deposit_event(Event::ERC20TransactionStored(
+					who.clone(),
+					*eth_tx_hash,
+					*eth_address,
+					*erc20_amount,
+				));
 			}
-			let end_height = EndHeight::get();
-			   if height > end_height {
-				EndHeight::put(height);
+			if let Some(end_height) = EndHeight::<T>::get() {
```

Comment:
Generally you can use `T::get().unwrap_or_default()` to mimic the substrate 2.0 behavior